app/app/src/main/kotlin/com/hedvig/app/feature/profile/ui/payment/PaymentDestination.kt

```diff
@@ -0,0 +1,321 @@
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedButton
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedSmallButton
+import com.hedvig.android.core.designsystem.component.card.HedvigCard
+import com.hedvig.android.core.designsystem.component.textfield.HedvigTextField
+import com.hedvig.android.core.designsystem.material3.typeContainer
+import com.hedvig.android.core.designsystem.material3.typeElement
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.icons.Hedvig
+import com.hedvig.android.core.icons.hedvig.normal.ChevronRight
+import com.hedvig.android.core.icons.hedvig.normal.Payments
+import com.hedvig.android.core.icons.hedvig.normal.Waiting
+import com.hedvig.android.core.ui.clearFocusOnTap
+import com.hedvig.android.core.ui.scaffold.HedvigScaffold
+import hedvig.resources.R
+import java.util.*
+import kotlinx.datetime.LocalDate
+
+@Composable
+fun PaymentDestination(navigateUp: () -> Unit) {
+
+}
+
+@Composable
+fun PaymentScreen(
+  uiState: PaymentUiState,
+  locale: Locale,
+  navigateUp: () -> Unit,
+) {
+  Box(
+    modifier = Modifier.fillMaxSize(),
+  ) {
+    HedvigScaffold(
+      topAppBarText = stringResource(R.string.PROFILE_PAYMENT_TITLE),
+      navigateUp = navigateUp,
+      modifier = Modifier.clearFocusOnTap(),
+    ) {
+      Column(modifier = Modifier.padding(horizontal = 16.dp)) {
+        Spacer(Modifier.height(16.dp))
+        NextPayment(uiState, locale)
+        Spacer(Modifier.height(4.dp))
+        Divider()
+        Spacer(Modifier.height(4.dp))
+        InsuranceCosts(uiState, locale)
+        Spacer(Modifier.height(4.dp))
+        TotalDiscount(uiState)
+        Spacer(Modifier.height(4.dp))
+        Divider()
+        Spacer(Modifier.height(16.dp))
+        AddDiscount(uiState)
+        Spacer(Modifier.height(32.dp))
+        PaymentDetails(uiState)
+        Spacer(Modifier.height(4.dp))
+        Divider()
+        PaymentHistory(onClick = {})
+        Spacer(Modifier.height(16.dp))
+        HedvigContainedButton(
+          text = stringResource(id = R.string.PROFILE_PAYMENT_CHANGE_BANK_ACCOUNT),
+          onClick = {},
+        )
+      }
+    }
+  }
+}
+
+@Composable
+private fun NextPayment(uiState: PaymentUiState, locale: Locale) {
+  HedvigCard(modifier = Modifier.fillMaxWidth()) {
+    Text(
+      text = uiState.nextChargeAmount.format(locale),
+      textAlign = TextAlign.Center,
+      style = MaterialTheme.typography.displayMedium,
+      modifier = Modifier.padding(vertical = 6.dp),
+    )
+  }
```

Comment:
Maybe this can be not part of the `NextPayment`, or maybe the Row below this? Or are these two components considered one in the design?

Reply:
I just though it made sense to group this since it contains both the next charge amount and the next charge date